napari/_qt/_qapp_model/qactions/_file.py

```diff
@@ -0,0 +1,169 @@
+    window._qt_viewer._open_files_dialog(choose_plugin=True)
+
+
+def _open_files_as_stack_with_plugin(window: Window):
+    window._qt_viewer._open_files_dialog_as_stack_dialog(choose_plugin=True)
+
+
+def _open_folder_with_plugin(window: Window):
+    window._qt_viewer._open_folder_dialog(choose_plugin=True)
+
+
+Q_FILE_ACTIONS: List[Action] = [
+    Action(
+        id=CommandId.DLG_OPEN_FILES,
+        title=CommandId.DLG_OPEN_FILES.title,
+        callback=QtViewer._open_files_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[StandardKeyBinding.Open],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FILES_AS_STACK,
+        title=CommandId.DLG_OPEN_FILES_AS_STACK.title,
+        callback=QtViewer._open_files_dialog_as_stack_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[{'primary': KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KeyO}],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FOLDER,
+        title=CommandId.DLG_OPEN_FOLDER.title,
+        callback=QtViewer._open_files_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.NAVIGATION}],
+        keybindings=[
+            {'primary': KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.KeyO}
+        ],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FILES_WITH_PLUGIN,
+        title=CommandId.DLG_OPEN_FILES_WITH_PLUGIN.title,
+        callback=_open_files_with_plugin,
+        menus=[
+            {'id': MenuId.FILE_OPEN_WITH_PLUGIN, 'group': MenuGroup.NAVIGATION}
+        ],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FILES_AS_STACK_WITH_PLUGIN,
+        title=CommandId.DLG_OPEN_FILES_AS_STACK_WITH_PLUGIN.title,
+        callback=_open_files_as_stack_with_plugin,
+        menus=[
+            {'id': MenuId.FILE_OPEN_WITH_PLUGIN, 'group': MenuGroup.NAVIGATION}
+        ],
+    ),
+    Action(
+        id=CommandId.DLG_OPEN_FOLDER_WITH_PLUGIN,
+        title=CommandId.DLG_OPEN_FOLDER_WITH_PLUGIN.title,
+        callback=_open_folder_with_plugin,
+        menus=[
+            {'id': MenuId.FILE_OPEN_WITH_PLUGIN, 'group': MenuGroup.NAVIGATION}
+        ],
+    ),
+    Action(
+        id=CommandId.DLG_SHOW_PREFERENCES,
+        title=CommandId.DLG_SHOW_PREFERENCES.title,
+        callback=Window._open_preferences_dialog,
+        menus=[{'id': MenuId.MENUBAR_FILE, 'group': MenuGroup.PREFERENCES}],
+        # TODO: revert to `StandardKeyBinding.Preferences` after app-model>0.2.0
+        keybindings=[{'primary': KeyMod.CtrlCmd | KeyCode.Comma}],
+    ),
+    # TODO:
+    # If app-model supports a `kwargs` field, remove the lambda in the callback.
+    # If it also allows registration of the same `id` when args are different,
+    # we can remove the f string in `id`
+    Action(
+        id=f"{CommandId.DLG_SAVE_LAYERS}.selected",
+        title=trans._('Save Selected Layers...'),
+        callback=lambda: QtViewer._save_layers_dialog(selected=True),
```

Comment:
@jni or @Czaki 

Is there another way (non lambda) way to deal with this that would be preferred?

If not, this raises a mypy error:

> napari/_qt/_qapp_model/qactions/_file.py:95: error: Missing positional argument "self" in call to "_save_layers_dialog"  [call-arg]

Would it be acceptable to ignore this error? Thanks

Reply:
tbh I don't know how this works even *before* mypy... Is `QtViewer._save_layers_dialog` a classmethod, or is this some app-model/in-n-out magic where the callable gets the actual QtViewer instance injected? ... In which case, I don't understand the use of an argument-less lambda, even before we get to mypy...

Reply:
@lucyleeow can you confirm that this actually *works* as intended at runtime? (ie before getting mypy involved)

Reply:
`QtViewer._save_layers_dialog` is a class method. 
This action used to be defined in `napari/_qt/menus/file_menu.py`:
```python
                'text': trans._('Save Selected Layer(s)...'),
                'slot': lambda: window._qt_viewer._save_layers_dialog(
                    selected=True
                ),
                'shortcut': 'Ctrl+S',
                'enabled': self._layer_count,
            },
```
Yes it does work.

Reply:
Whoops, the lambda does not work, but `QtViewer._save_layers_dialog` below does, sorry